### PR TITLE
Add Functionality: Show next upcoming episode

### LINKED ISSUE #301: [ENHANCEMENT] - Show upcoming episode on homepage
Based upon the great work done to have a list of upcoming episodes, it would be brilliant to have an option to adjust the jumbotron on the index page to show the next upcoming episode (or perhaps 2-3 at the discretion of configuration)?

### PR DESCRIPTION
**Pull request by chrisreddington**
There is an opportunity to "half" the screen real estate of the Most Recent episode section, and also use it as an opportunity to showcase the next episode that is due to be released.

This PR implements that with some logic to check if - 
1. Does enable_show_next_episode exist in the config?
2. Is enable_show_next_episode set to true in the config?
3. Are there upcoming episodes to show?

If all 3 conditions are matched, then the row partial will split the top section into two, allowing to show the next upcoming episode.

Fixes #301

### CODE REVIEW

**Comment by chrisreddington** [comment]
@mattstratton - Right now, this doesn't implement any changes to the grid view. I wasn't sure how best to handle this scenario for the grid view, as there's no real concept of "latest" there...

One way could be to adjust the query in the grid view to also show the next upcoming episode, but it might not be as clear as it is in the row view. So, i'm curious on your thoughts on this one? Thx!

**Comment by mattstratton** [comment]
> Right now, this doesn't implement any changes to the grid view. I wasn't sure how best to handle this scenario for the grid view, as there's no real concept of "latest" there...

Yeah, I suspect this feature would only work for the row layout anyway :)

File: REFERENCE.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ These should be set under the `[params]` section:
...
 | `enable_jumbo`          | No       | When using the `row` layout, will set a jumbotron at the top instead of the sidebar.                                                                                                                                                                                | "true"                                                                                                                                                                                                                                    |
+| `enable_show_next_episode`          | No       | When using the `row` layout, if there are episodes with the upcoming frontmatter set to true, then it will display the next episode on homepage. This halves the screen real estate of the latest episode section, which will then be displayed side by side. | "true"                                                                                                                                                                                                                       |
```
Comment (mattstratton) [inline REFERENCE.md:47]: I wonder if we could give this a more descriptive name, like `show_next_upcoming` which makes it more clear that it is only for upcoming?

**Comment by mattstratton** [comment]
I took a screenshot of what it looks like when enabled (for reference)

image

I think that we would want to do some styling on the "upcoming" episode; maybe draw a box around it with some other shading/bg color just to set it off?

Another option is to have less info for the "next episode" and make it a row by itself, but not include any images, etc, but just like a row across that says like "Next episode! Awesome Tech with George Bluth on 7 Dec 2020!" or something? If we did some kind of hero type row like that (narrow in my mind) it would work on grid too.

**Comment by mattstratton** [comment]
I am imagining something like this (rough mockup) 
upcoming

**Comment by mattstratton** [comment]
btw, if you sign off your commits, the DCO bot will stop flagging your PRs/commits as failed :)

https://github.com/mattstratton/castanet/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco

File: REFERENCE.md (hunk 1 above)
Comment (chrisreddington) [reply REFERENCE.md:47]: Good call, I'll make that change

**Comment by chrisreddington** [comment]
@mattstratton - So this would work? :) (The name of the episode is a link, of course)

image

**Comment by mattstratton** [comment]
yes! that looks good. And should be a way to do it in grid view now too!

**Comment by mattstratton** [comment]
You might want to rebase this branch since I just merged a couple things, one of which touches REFERENCE.MD. It's probably a clean merge anyway but if you feel like rebasing it wouldn't hurt.

**Comment by chrisreddington** [comment]
@mattstratton - Rebased, and committed. Also signed off the commits (thanks for the tip!)

Believe this is now ready for review.

**Comment by mattstratton** [comment]
Aha! it's a little trickier than that to edit the CSS files; you have to actually edit the SCSS files and re-compile them with `gulp`. There's instructions in the `CONTRIBUTING.md` file.

Long story short, yes, editing them by hand will work, but the next time they are compiled your changes will get blown away. 

the good news is you really only need to edit the `custom.scss` file and as long as you use variable names, it'll update for the various color themes.

**Comment by mattstratton** [comment]
If the SCSS stuff is too tricky, let me know and I think I can checkout this branch/PR and make the changes for you!

**Comment by mattstratton** [comment]
Since the colors are the same for all the themes, I'm just going to do the `gulp` build for you and I'll push up the changes.

Taking me longer than I thought since the version of `node-sass` I've been using doesn't work on Big Sur :)

**Comment by chrisreddington** [comment]
Hadn't spotted the scss in there, my bad - but good to know for future contributions :)

**Comment by mattstratton** [comment]
ugh. need to do some updates for node, etc...

**Comment by mattstratton** [comment]
okay, I think I got the build happy now! This looks clean and I'm going to merge it and cut a release.

**Pull request merged**